### PR TITLE
Survivor Variant Picking + Survivor Priority (Reopened Edition)

### PR DESCRIPTION
**Pull request by Unknownity**
# About the pull request

Re-opening #1614 with the permission of Geeves. Resolved conflicts and made changes of all the requested reviews that were made.

This might need testmerging first.

# Explain why it's good for the game

"It makes it easier to make a survivor character to actually RP with if you can write their background (medical, security, etc), and it also makes special roles appear more often if the Nightmare spawn actually spawns."

# Testing Photographs and Procedure
From my general testing everything that would gotten affected works so far with no reported runtimes. One anomaly reported is if you force a round to start, you need to do it twice for some reason.

# Changelog

:cl: Geeves, Unknownity
add: Added survivor variant picking, in the character creation menu, under special roles. This makes it so you only spawn as that survivor type, unless you hit a special spawn, such as CLF or FORECON, for example.
add: Added survivor spawner priority, so that the CLF and PMC spawners are used before others.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
fix your CL you reopened this, make it 'Geeves, Uknownitiytiy'

File: code/__DEFINES/job.dm [hunk 1]
```diff
@@ -30,6 +30,22 @@ var/global/list/job_squad_roles = JOB_SQUAD_ROLES_LIST
...
+#define ENGINEERING_SURVIVOR			"Engineering Survivor"
+#define CORPORATE_SURVIVOR				"Corporate Survivor"
+#define SURVIVOR_VARIANT_LIST			list(ANY_SURVIVOR, CIVILIAN_SURVIVOR, SECURITY_SURVIVOR, SCIENTIST_SURVIVOR, MEDICAL_SURVIVOR, ENGINEERING_SURVIVOR, CORPORATE_SURVIVOR)
+
+#define SPAWN_PRIORITY_VERY_HIGH		1
+#define SPAWN_PRIORITY_HIGH				2
+#define SPAWN_PRIORITY_MEDIUM			3
+#define SPAWN_PRIORITY_LOW				4
+#define SPAWN_PRIORITY_VERY_LOW			5
+#define LOWEST_SPAWN_PRIORITY			5
```
Comment (carlarctg) [inline code/__DEFINES/job.dm:47]:
```suggestion
#define SPAWN_PRIORITY_VERY_LOW			5
#define LOWEST_SPAWN_PRIORITY			SPAWN_PRIORITY_VERY_LOW
```

Comment (Unknownity) [reply code/__DEFINES/job.dm:47]: the defines are not supposed to be the same, while the numbers are the same, these can be changed in the future if someone decides to.

**Review by harryob — APPROVED** [review]
despite that ternary giving me the ick

**Pull request merged**